net: unify FAR keyword usage for all net buffer memory mapped buffers

pkarashchenko opened the pull request:
## Summary
In some cases the `FAR` keyword is missing when working with network buffers and using memory mapped types. Restore consistency

## Impact
No functional changes

## Testing
Pass CI

gustavonihei reviewed (changes requested):
FAR qualifier is only required for common code. Many (if not all, haven’t looked at all files) are restricted to chip implementation.
Besides, FAR and CODE have already been removed previously from Xtensa and RISC-V before. Adding them back would be a step back.

pkarashchenko commented:
> FAR qualifier is only required for common code. Many (if not all, haven’t looked at all files) are restricted to chip implementation. Besides, FAR and CODE have already been removed previously from Xtensa and RISC-V. Adding them back would be a step back.

@gustavonihei I haven't added neither `FAR` nor `CODE` to Xtensa and RISC-V.
For each file that I added I examined if `FAR` is already used in that file or not and if I found `FAR` then I added it as well.

I think that `FAR` and `CODE` keywords removal should be by code for each architecture step by step in all arch dependent places. I just try to keep consistency with existing code.

gustavonihei commented:
> > FAR qualifier is only required for common code. Many (if not all, haven’t looked at all files) are restricted to chip implementation. Besides, FAR and CODE have already been removed previously from Xtensa and RISC-V. Adding them back would be a step back.
> 
> @gustavonihei I haven't added neither `FAR` nor `CODE` to Xtensa and RISC-V. For each file that I added I examined if `FAR` is already used in that file or not and if I found `FAR` then I added it as well.
> 
> I think that `FAR` and `CODE` keywords removal should be by code for each architecture step by step in all arch dependent places. I just try to keep consistency with existing code.

Sorry, @pkarashchenko. I had a quick look on the cellphone and misunderstood some bits of the PR. I'll drop the review for now.

xiaoxiang781216 commented on arch/mips/src/pic32mz/pic32mz_ethernet.c:
```diff
@@ -254,7 +254,7 @@
  */
 
-#define BUF ((struct eth_hdr_s *)priv->pd_dev.d_buf)
+#define BUF ((FAR struct eth_hdr_s *)priv->pd_dev.d_buf)
```
remove

pkarashchenko replied:
Currently `FAR` is heavily used in arch/mips files

xiaoxiang781216 replied:
Ok, let's keep it

xiaoxiang781216 commented on arch/mips/src/pic32mx/pic32mx_ethernet.c:
```diff
@@ -215,7 +215,7 @@
  */
 
-#define BUF ((struct eth_hdr_s *)priv->pd_dev.d_buf)
+#define BUF ((FAR struct eth_hdr_s *)priv->pd_dev.d_buf)
```
remove

pkarashchenko replied:
Currently `FAR` is heavily used in arch/mips files

xiaoxiang781216 replied:
Ok, let's keep it.

pkarashchenko commented on arch/xtensa/src/esp32/esp32_wlan.c:
```diff
@@ -1077,7 +1077,7 @@ static void wlan_txtimeout_expiry(wdparm_t arg)
 static void wlan_poll_work(void *arg)
 {
   int32_t delay_tick = WLAN_WDDELAY;
-  FAR struct wlan_priv_s *priv = (FAR struct wlan_priv_s *)arg;
```
Yes. The `FAR` is not used anywhere for xtensa
